subcommands/add-service

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+source "$(dirname "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)")/config"
+set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
+source "$PLUGIN_BASE_PATH/common/functions"
+source "$(dirname "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)")/internal-functions"
+
+cmd-acl-add-service() {
+  #E allow the servuser user access to the birds Redis service
+  #E dokku $PLUGIN_COMMAND_PREFIX:add-service redis birds servuser
+  #A type, type of service
+  #A service, service to run command against
+  #A user, a user to allow access
+  declare desc="allow <user> to control <service> of type <type>"
+  local cmd="$PLUGIN_COMMAND_PREFIX:add-service" argv=("$@"); [[ ${argv[0]} == "$cmd" ]] && shift 1
+  declare SERVICE_TYPE="$1" SERVICE="$2" USER="$3"
+
+  fn-acl-check-service "$SERVICE_TYPE" "$SERVICE"
+
+  local ACL_PATH="$DOKKU_LIB_ROOT/services/$SERVICE_TYPE/$SERVICE/acl"
+  local ACL_FILE="$ACL_PATH/$USER"
+
+  if [[ -z "$USER" ]]; then
+    dokku_log_fail "Please specify a user name"
+  fi
+
+  [[ ! -d "$ACL_PATH" ]] && mkdir -p "$ACL_PATH"
```

Comment:
We should enforce that this is owned by `DOKKU_SYSTEM_USER` AND `DOKKU_SYSTEM_GROUP`. Please note that this won't be defaulted to `dokku` until the next release, so we can just set that as a default in this script before usage like so:

```
export DOKKU_SYSTEM_USER=${DOKKU_SYSTEM_USER:="dokku"}
export DOKKU_SYSTEM_GROUP=${DOKKU_SYSTEM_GROUP:="dokku"}
```

Reply:
What do you mean by "enforce"? Check ownerships and fail if it's something unexpected? Check ownerships when we check the ACL? Both of these would be new features not already done by per-app commands and hooks, so I don't think they're a good fit for this PR.

Reply:
IIRC these commands can only be run as `root`, right? If thats the case, we'll actually run the cli on the server as the `dokku` user - it drops perms to that before doing anything else in the `dokku` bin - and the perms on the folder/files won't be correct so accessing it will fail.

If this works otherwise, cool, but I've seen this fail in weird ways because of perms. Better safe than sorry I guess?

Reply:
You can run the `acl:*` commands as the `dokku` user too, if you have access to them. (Generally if dokku-acl is in use, users won't have access to `acl:*` but I suppose someone could add those commands to the dokku-acl config.)

Whether they're run as `dokku` or `root`, the files end up being owned by `dokku:dokku`:

```
root@be09e07473d9:/app# dokku acl:list-service redis foo6
admin
root@be09e07473d9:/app# ls -ld /var/lib/dokku/services/redis/foo6/acl
drwxr-xr-x 2 dokku dokku 4096 Jul 24 21:30 /var/lib/dokku/services/redis/foo6/acl
root@be09e07473d9:/app# ls -l /var/lib/dokku/services/redis/foo6/acl
total 0
-rw-r--r-- 1 dokku dokku 0 Jul 24 21:30 admin
```

I'm still unclear on what you want to do here. I think it's best to make any changes as part of a separate PR so they can be made to existing commands as well, but if you want changes I'll log a public issue so I don't forget.

Reply:
Ah didn't realize the allow/deny stuff is available over ssh. I guess we only block that for the `plugin` subcommands (which makes sense since that is in the core...).

This is good to merge as is then :)